reproducible/faiss/Dockerfile

```diff
@@ -0,0 +1,45 @@
+FROM docker.io/library/debian:11
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Phase 1: Install free components.
+RUN apt-get update
+RUN apt-get install -y --no-install-recommends \
+    python3-dev python3-numpy-dev swig python3-virtualenv \
+    cmake git gpg wget
+
+# Phase 2: Install non-free components.
+RUN sed -i -e 's/main/main contrib non-free/g' /etc/apt/sources.list
+
+# Add nVidia CUDA repository.
+RUN wget https://developer.download.nvidia.com/compute/cuda/repos/debian11/x86_64/cuda-keyring_1.1-1_all.deb
+RUN dpkg -i cuda-keyring_1.1-1_all.deb
+
+# Add Intel repository.
+RUN wget -O- https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB \
+  | gpg --dearmor | tee /usr/share/keyrings/oneapi-archive-keyring.gpg > /dev/null
+RUN echo "deb [signed-by=/usr/share/keyrings/oneapi-archive-keyring.gpg] https://apt.repos.intel.com/oneapi all main" \
+  >> /etc/apt/sources.list.d/oneAPI.list
+
+RUN apt-get update
+RUN apt-get install -y --no-install-recommends \
+    cuda intel-basekit
```

Comment:
could this not be all one line?

Reply:
Yeah, but it takes like 20min for me to run line 25 on my residential Comcast. Chonky packages.

My professional friend Itamar has [a sweet article](https://pythonspeed.com/articles/system-packages-docker/) on how to do this the right way. I just don't have bandwidth for it.

Reply:
@MostAwesomeDude I am unsure how 25 and 26 being on separate lines with a `\` changes the build time. I do understand the article (the link through has a reference to multistage builds, which are easy to use and could be a further improvement is needed.